onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -308,24 +309,28 @@ def destroy(self, request, *args, **kwargs):
 
                 if not instance_ids:
                     raise ParseError(_(u"Invalid data ids were provided."))
-
                 initial_count = self.object.submission_count()
-                queryset = Instance.objects.filter(
+
+                Instance.objects.filter(
                     id__in=instance_ids,
```

Comment:
You should handle cases where the user wants to delete all the form submissions. In that case the user can just pass a `delete_all` flag  and this query will change to  `Instance.objects.filter(xform=self.object ...`

Reply:
handled this use case. thanks